### PR TITLE
Actually resize avatars

### LINKED ISSUE #770: Avatars should be resized to 512x512
## Expected Behavior

Avatars should be resized to 512x512.

## Current Behavior

At least when setting the profile, the avatar is not being resized and sent as-is to the app.

### PR DESCRIPTION
**Pull request by lgrahl**
Formerly, avatars have not been resized. Updated croppie type definitions so this doesn't happen again.

However, the race condition is still unfixed: When the user clicks on "Ok" before the timeout of 500ms fired, the avatar will not be updated. @dbrgn I have a patch ready if you see an easy way to communicate between the two controllers.

Resolves #770

### CODE REVIEW

File: src/directives/avatar_editor.ts [hunk 1]
```diff
@@ -41,47 +44,41 @@ export default [
...
+                                        circle: false,
+                                        format: 'png',
                                     });
-                            }, 500);
-                        },
-                    });
-
+                                    const fileReader = new FileReader();
+                                    fileReader.onload = function() {
+                                        scope.onChange(fileReader.result);
```
Comment (dbrgn) [inline src/directives/avatar_editor.ts]: Just as a sidenote: `this` and `fileReader` are equivalent in the `onload` function, but I agree that using `fileReader` makes things more explicit.

Comment (lgrahl) [reply src/directives/avatar_editor.ts]: They aren't the same with the refactoring any more. But yeah, let's avoid the brittleness of implicit `this` in general.

**Comment by lgrahl** [comment]
I'll double-check that.

**Comment by lgrahl** [comment]
`newImage(null)` is never being called.

**Pull request merged**